apps/extension/src/core/util/hasErc721Nft.ts

```diff
@@ -0,0 +1,38 @@
+import { AccountType } from "@core/domains/accounts/types"
+import { chainConnectorEvm } from "@core/rpcs/chain-connector-evm"
+import keyring from "@polkadot/ui-keyring"
+import { EvmNetworkId } from "@talismn/chaindata-provider"
+import { parseAbi } from "viem"
+
+import { abiErc721 } from "./abi"
+
+type Address = `0x${string}`
+
+export const hasErc721Nft = async ({
+  evmNetworkId,
+  contractAddress,
+}: {
+  evmNetworkId: EvmNetworkId
+  contractAddress: Address
+}): Promise<Record<Address, boolean>> => {
+  const evmAddresses = keyring
+    .getPairs()
+    .filter(({ type, meta }) => type === "ethereum" && meta.origin !== AccountType.Watched)
+    .map(({ address }) => address)
+
+  const client = await chainConnectorEvm.getPublicClientForEvmNetwork(evmNetworkId)
+  if (!client) throw new Error(`Unable to connect to EVM network: ${evmNetworkId}`)
+
+  const data = await client.multicall({
```

Comment:
multicall is automatic for all contract calls if network supports it, so here i think we should use a Promise.all([contract calls]) : 
- this method is network agnostic, and not all network support multicall.
- it would preserve typings for the results

Reply:
ex here : https://github.com/TalismanSociety/talisman/blob/44a1066ee5e907844650ac39919c3fcf5f5617f9/apps/extension/src/core/util/decodeEvmTransaction.ts#L90